src/negotiation/order.ts

```diff
@@ -16,8 +16,125 @@ export interface OrderAsset {
   nominalAmount: string;
 }
 
+export interface TakerCriteria {
+  buy: TakerCriteriaAsset;
+  sell: TakerCriteriaAsset;
+  minRate?: number;
+}
+
+export interface TakerCriteriaAsset {
+  ledger: string;
+  asset: string;
+  minNominalAmount?: string;
+  maxNominalAmount?: string;
+}
+
+/**
+ * @description An order handler for the taker. It has helper functions to facilitate the handler of an
+ * order by a taker. This should only be created via `TakerNegotiatior.getOrder()` and should not be constructed from
```

Comment:
🙃 If `TakerNegotiatior.getOrder()` creates an `Order` it should probably be renamed to `TakerNegotiatior.getNewOrder()`

Reply:
Not sure I agree here as the order comes from the order book and hence is not "new" per se.